src/middlewares/userMiddlewares.js

```diff
@@ -0,0 +1,29 @@
+import models from '../models';
+
+/**
+ * User Middlewares Class
+ */
+export default class UserMiddlewares {
+/**
+ * @name checkUserExists
+ * @description Checks if a user exists in the database
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @param {object} next The response object
+ * @returns {object} The API response or next()
+ */
+  static async checkUserExists(req, res, next) {
+    try {
+      const { email } = req.body;
+      const data = await models.User.findOne({ where: { email, } });
+      console.log(data);
+      if (!data) return next();
+      return res.status(409).json({
+        status: 409,
+        error: 'Unsuccesful, user already exists, kindly use a different email and/or username.',
```

Comment:
Are we making userName and email unique

Reply:
what do you suggest?, only email should be unique?